src/main/java/org/eclipsefoundation/marketplace/helper/JavaVersionHelper.java

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.helper;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Used to handle the conversion of Java versions to and from storage mediums.
+ * 
+ * @author Martin Lowe
+ *
+ */
+public class JavaVersionHelper {
+	private static final Pattern MAJOR_VERSION_CHECK = Pattern.compile("^\\d\\.([^\\.\\D]+)");
+
+	/**
+	 * Converts value to strip out parts of Java version string that cause issues
+	 * with natural number sorting in databases.
+	 * 
+	 * @param val the java version string to convert (e.g. 1.7, 1.10)
```

Comment:
I think this comment is misleading. I'm pretty sure that started with Java 9, you will never see 1.9, 1.10 or 1.11 but only 9, 10 or 11.